lm_eval/utils.py

```diff
@@ -104,12 +149,51 @@ def complete_code(
             for sample, generated_tokens in zip(generated_tasks, generated_tokens):
                 gen_token_dict[sample].append(generated_tokens)
 
+    def strip_left_padding(tokens, pad_token_id):
+        """Strip left padding from tokens."""
+        for i, token in enumerate(tokens):
+            if token != pad_token_id:
+                return tokens[i:]
+        return tokens
+
+    def parse_infill(code, tokenizer):
```

Comment:
I think we should probably return only the middle part instead or reodering and returning all code. Because we can get the latter by concatenating with prefix & suffix in the Task file, but the opposite is not always possible (e.g when there are no delimiters in the code itself like [FIM](https://github.com/nuprl/MultiPL-E/blob/c68410b5655bf80f8f49fa802de4cc7f5823d34d/inference/incoder.py) exact match from HumanEval/MuliPL-E which requires the middle part)

Reply:
Good point, thanks for raising. Resolved [here](https://github.com/bigcode-project/bigcode-evaluation-harness/pull/39/commits/2fc687534741496a886c7d877ffdf456799e99a1).